app/models/user.rb

```diff
@@ -8,16 +8,32 @@ class User < BaseModel
 
   has_one :address, as: :addressable
   has_one :image, -> { where(parent_id: nil) }, class_name: 'UserImage'
-  has_one :individual_account, foreign_key: 'opener_id'
 
   has_many :memberships
-  has_many :member_accounts, through: :memberships, source: :account
+  has_many :accounts, through: :memberships, source: :account
   has_many :producers
 
-  def accounts
-    Account.
-      joins('LEFT OUTER JOIN `memberships` ON `memberships`.`account_id` = `accounts`.`id`').
-      where(['accounts.id = ? OR memberships.user_id = ?', individual_account.id, id])
+  after_commit :create_individual_account, on: [:create]
+
+  def individual_account
+    accounts.where('type = \'IndividualAccount\'').first
+  end
+
+  def individual_account=(account)
+    prior_account = individual_account
+    memberships.create!(account_id: account.id, approved: true, role: 'admin')
+    if prior_account
+      memberships.where(account_id: prior_account.id).each { |m| memberships.destroy(m) }
```

Comment:
In what case would a user have a prior individual account, that needed to be destroyed?